main/boofcv-ip/src/main/java/boofcv/alg/color/ColorDiff.java

```diff
@@ -0,0 +1,240 @@
+package boofcv.alg.color;
+
+/**
+ * ColorDiff methods.
+ *
+ * The best source for color difference  is [3]. I believe it is the most correct implementation I've found. Usually
+ * different sources shows different results, especially for CIEDE2000. [1] is used to implement CIEDE2000.
+ * [3] used to implement remaining functions. [2] is a source of comparative data for RGB delta difference,
+ * although some results are still little different and I could not found the source of error.
+ * [4] is also a good source of information.
+ *
+ * Equation from:
+ * [1] Sharma G, Wu W, Dalal EN.The CIEDE2000 Color-Difference Formula: Implementation Notes, Supplementary Test
+ *      Data and Mathematical Observations. 2004.
+ * [2] Mokrzycki WS, Tatol M. Colour difference delta E - A survey
+ * [3] http://brucelindbloom.com/index.html?ColorDifferenceCalc.html
+ * [4] https://en.wikipedia.org/wiki/Color_difference
+ *
+ * @author  Jairo Rotava
+ *
+ */
+public class ColorDiff {
+    final static int l = 0;
```

Comment:
Could you add a line line comment say what these are?

Reply:
they are index to the L, a and b values in the lab array. just to get rid the 0, 1 and 2 indexes. Will comment